### ISSUE TITLE
Upgrade Polly dependency to Polly v5.9.0

### ISSUE THREAD

**Issue by reisenberger** [aspnet/HttpClientFactory#92]
[Polly v5.9.0](https://www.nuget.org/packages/Polly/5.9.0) is published.  PR will follow to update HttpClientFactory to reference Polly v5.9.0.

**TL;DR** Polly v5.9.0 includes some deprecations of APIs we will rationalise/rename in Polly v6.0.  None of these changes affect `HttpClientFactory` at 2.1, but adopting Polly v5.9.0 in HttpClientFactory now, before ASP.NET Core 2.1 RTM, avoids any delta in the wider Polly experience for HttpClientFactory users between ASP.NET Core 2.1 and v-next.

Per semver and ASP.NET Engineering Guidelines, Polly v5.9.0 adds the replacement APIs (where relevant) and uses `[Obsolete("with an explanation here of the alternative API to be used, where relevant")]` to deprecate the APIs which will be removed at Polly v6.0.

---

#### Principal changes with relevance to HttpClientFactory

Polly v5.8.0 -> v5.9.0:

[1] [rationalises Execute/Async(,...) overloads](https://github.com/App-vNext/Polly/issues/425) by deprecating legacy/little-used and anomalous overloads.
  - No impact for `HttpClientFactory` or consumers of `HttpClientFactory` as `PolicyHttpMessageHandler` [abstracts away](https://github.com/aspnet/HttpClientFactory/blob/b9361570c6382a12a4733f785249e29929d4bca5/src/Microsoft.Extensions.Http.Polly/PolicyHttpMessageHandler.cs#L125) the `policy.ExecuteAsync(...)` call in any case.

[2] [clarifies naming of Polly telemetry properties](https://github.com/App-vNext/Polly/issues/426) in preparation for richer telemetry/logging from Polly executions.
  - No impact for `HttpClientFactory` as consumption of these properties for logging is not part of ASP.NET Core 2.1 release.

[3] [improves syntax](https://github.com/reisenberger/HttpClientFactory/commit/d2303d39976fabcab9b9431dedd4fb3f4a3678b9#diff-4f6d37142cd8d6cdc643811870d956adR44) for the [edge case](https://github.com/App-vNext/Polly/wiki/Non-generic-and-generic-policies#when-you-hold-a-non-generic-policy-but-need-a-generic-policy) where users configure a non-generic `Policy` such as `Policy.TimeoutAsync(10);` but wish to use it with `HttpClientFactory`; for HttpClientFactory the decision was to [accept generic `IAsyncPolicy<HttpResponseMessage>` only](https://github.com/aspnet/HttpClientFactory/blob/b9361570c6382a12a4733f785249e29929d4bca5/src/Microsoft.Extensions.Http.Polly/PolicyHttpMessageHandler.cs#L83).

+ only affects [explanatory XML remarks within HttpClientFactory](https://github.com/reisenberger/HttpClientFactory/commit/d2303d39976fabcab9b9431dedd4fb3f4a3678b9#diff-4f6d37142cd8d6cdc643811870d956adR44); change will be PR'd shortly.

---

Should further detail be required, every change in Polly v5.9.0 is milestone-tagged [here](https://github.com/App-vNext/Polly/milestone/26?closed=1) and [here](https://github.com/App-vNext/Polly/milestone/26).

**Comment by reisenberger** [comment]
@rynowak @glennc /cc @joelhulen 

We have rarely made breaking changes in Polly.  We are taking the opportunity of a v6.0 to rationalise and clarify two sections of the API.  Getting these changes into HttpClientFactory now avoids delta in experience for users after ASP.NET Core 2.1.

**Comment by rynowak** [comment]
@Eilon  - can you please do the OSS homework for this?

**Comment by Eilon** [comment]
@rynowak : done: https://github.com/aspnet/HttpClientFactory/pull/93#pullrequestreview-110999582

**Comment by rynowak** [comment]
I've merged #93 - is there any other action we need to take? It wasn't clear if point 3 was addressed yet

**Comment by reisenberger** [comment]
Thanks @rynowak @Eilon (aiming to keep to [Engineering Guidelines](https://github.com/aspnet/Home/wiki/Engineering-guidelines) to keep this all smooth)

@rynowak Yes, the change for [3] was in the PR you merged; change can be seen in dev branch now [here](https://github.com/aspnet/HttpClientFactory/blob/60a23380a354ebaf27c8cd3505e745bd8b7f1541/src/Microsoft.Extensions.Http.Polly/PolicyHttpMessageHandler.cs#L44).  No further action arising from this issue #92

**Comment by rynowak** [comment]
Great, closing this out then. Thanks again for the help!
